File: src/server/src/main/java/com/project/application/models/exercise/EquipmentResource.java
package com.project.application.models.exercise;

import com.project.domain.workout.Equipment;
import io.micronaut.serde.annotation.Serdeable;

import java.util.UUID;

@Serdeable
public record EquipmentResource(UUID id, String name) {
    public static EquipmentResource from(Equipment equipment) {
        return new EquipmentResource(equipment.getId(), equipment.getName());
    }
}

File: src/server/src/main/java/com/project/application/models/exercise/EquipmentResource.kt
package com.project.application.models.exercise

import com.project.domain.workout.Equipment
import io.micronaut.serde.annotation.Serdeable
import java.util.UUID

@Serdeable
data class EquipmentResource(val id: UUID, val name: String?) {
    companion object {
        fun from(equipment: Equipment): EquipmentResource {
            return EquipmentResource(equipment.id, equipment.name)
        }
    }
}
